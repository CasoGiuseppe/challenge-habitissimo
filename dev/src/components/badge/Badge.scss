.badge {
  @include UI-base-rounded;
  @include UI-spinner(($base * 4), 3, $c-white);
  $badge: &;

  background: $c-secondary;
  padding: #{$base * 3px};
  position: relative;
  overflow: hidden;
  width: 100%;

  &::before,
  &::after {
    content: '';
    display: none;
    position: absolute;
    pointer-events: none;
    z-index: 3;
  }

  &::before {
    background: rgba($c-black, .5);
    height: 100%;
    left: 50%;
    top: 50%;
    width: 100%;
    transform: translateY(-50%) translateX(-50%);
  }

  &:after {
    display: none;
    top: 50%;
    left: 50%;
    margin-top: #{($base * 2) * -1px};
    margin-left: #{($base * 2) * -1px};
  }

  &--isLoading {
    &::before,
    &::after {
      display: block;
    }
  }

  @include breakpoint('large-desktop') {
    padding: #{$base * 4px} #{$base * 6px};
  }

  &__title,
  &__subtitle,
  &__message {
    color: $c-white;
    padding-left: #{($base * 4px)};
  }

  &__title {
    @include text--bold;
    @include text--medium;
    position: relative;

    &::before {
      border-radius: 100%;
      background: $c-white;
      content: '';
      left: 0;
      height: #{$base * 2px};
      margin-right: #{$base * 2px};
      position: absolute;
      top: 0;
      transform: translateY(#{$base * 1px});
      width: #{$base * 2px};
    }

    + #{$badge}__subtitle {
      margin-top: #{$base * 1px};
    }

    + #{$badge}__message {
      margin-top: #{$base * 4px};
    }
  }

  &__subtitle {
    @include text--regular;
    @include text--small;

    + #{$badge}__message {
      margin-top: #{$base * 4px};

      @include breakpoint('only-phone') {
        margin-top: #{$base * 2px};
      }
    }
  }

  &__message {
    @include text--light;
    @include text--small;
    overflow: hidden;
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical; 
  }
}