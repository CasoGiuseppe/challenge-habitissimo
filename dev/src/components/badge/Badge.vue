<template>
    <article
      :class="[
        'badge',
        isLoading ? 'badge--isLoading' : null,
      ]">
        <h2
          v-if="$slots.title"
          class="badge__title">
          <slot name="title" />
        </h2>
      <h3
        v-if="$slots.subtitle"
        class="badge__subtitle">
        <slot name="subtitle" />
      </h3>
      <p
        v-if="$slots.message"
        class="badge__message">
        <slot name="message" />
      </p>
    </article>
</template>

<script>
export default {
  name: 'Badge',

  props: {
    id: {
      type: String,
    },

    isLoading: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang='scss' scoped src= './Badge.scss'></style>
