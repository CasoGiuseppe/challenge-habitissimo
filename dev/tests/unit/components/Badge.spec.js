import Badge from '@/components/badge/Badge'
import { mount } from '@vue/test-utils'

describe('Badge.vue', () => {
  let component

  beforeEach(() => {
    component = mount(Badge, {
      propsData:{
        id: 'test',
        isLoading: true,
        isLight: false,
        arrow: {
          dir: 'dir'
        },
        hasIcon: 'icon',
        isAnimated: true
      },
      slots: {
        image: '<span class="slot-image">image</span>',
        action: '<span class="slot-action">title</span>'
      }
    })
  })

  it('has correct props values', () => {
    expect(component.props().id).toBe('test')
    expect(component.props().isLoading).toBeTruthy()
    expect(component.props().isLight).toBeFalsy()
    expect(component.props().arrow).toHaveProperty('dir', 'dir')
    expect(component.props().hasIcon).toBe('icon')
    expect(component.props().isAnimated).toBeTruthy()
  })

  it('has correct action slot', () => {
    const main = component.find('.badge')
    expect(main.findAll(".slot-image").length).toBe(1);
    expect(main.findAll(".slot-action").length).toBe(1);
  })

  it('has correct class name', () => {
    expect(component.classes()).toContain('badge--isLoading')
  })

  it('has the expected html structure', () => {
    expect(component.html()).toMatchSnapshot()
  })
})