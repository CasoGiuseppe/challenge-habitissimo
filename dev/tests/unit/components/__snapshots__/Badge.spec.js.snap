// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge.vue has the expected html structure 1`] = `
<article class="badge badge--isLoading badge--hasIcon badge--hasArrow-undefined badge--hasImage"><span class="badge__image"><span class="slot-image">image</span></span>
  <transition-stub name="change-fade" mode="out-in">
    <div class="badge__enter-key">
      <!---->
      <!---->
      <!---->
      <!---->
      <div class="badge__action"><span class="slot-action">title</span></div>
    </div>
  </transition-stub>
</article>
`;
